core/orchestrator.go

```diff
@@ -359,7 +359,7 @@ func (orch *orchestrator) priceInfo(sender ethcommon.Address, manifestID Manifes
 
 		// If no priced capabilities were signaled by the broadcaster assume that they are requesting
 		// transcoding and set the base price to the transcode price
-		if basePrice.Cmp(big.NewRat(0, 1)) == 0 {
+		if transcodePrice != nil && (basePrice == nil || basePrice.Cmp(big.NewRat(0, 1)) == 0) {
```

Comment:
@eliteprox, as you mentioned earlier, `basePrice` will never be `nil` because of the following line in the `orchestrator.go` file:

[Orchestrator code](https://github.com/livepeer/go-livepeer/blob/a1b9f33d7261b798d0d33685992a4f93ec4d3473/core/orchestrator.go#L340).

Given this, I believe the condition `basePrice == nil` should be removed. This will ensure that the `basePrice` is correctly set to the `transcodePrice` when `basePrice` is `0`.

Here is the updated code snippet:

```suggestion
		if transcodePrice != nil && basePrice.Cmp(big.NewRat(0, 1)) == 0 {
```

This change ensures that the `basePrice` will be set to `transcodePrice` when it is zero, reflecting the intended behavior more accurately.